tests/unit/exceptions.test.ts

```diff
@@ -0,0 +1,80 @@
+describe("globalThis", () => {
+  it("globalThis should have a DOMException", () => {
+    expect(globalThis.DOMException()).toBeDefined();
+  });
+  it("globalThis.DOMException() should have a message", () => {
+    expect(globalThis.DOMException().message).toBeDefined();
+  });
+  it("globalThis.DOMException() should have a message", () => {
+    expect(globalThis.DOMException().name).toBeDefined();
+  });
+});
+
+describe("DOMException()", () => {
+  const e = new DOMException();
+
+  it("should have a message", () => {
+    expect(e.message).toBeDefined();
+  });
+  it("message property should be the initial value", () => {
+    expect(e.message).toEqual("");
+  });
+  it("should have a message", () => {
+    expect(e.name).toBeDefined();
+  });
+  it("should be the initial value", () => {
+    expect(e.name).toEqual("Error");
+  });
+  it("result of the toString method should be 'Error'", () => {
+    expect(e.toString()).toEqual("Error");
+  });
+  it("throwing and catching exceptions can be done", () => {
+    try {
+      throw new DOMException();
+    } catch (ex) {
+      expect(ex.toString()).toEqual(e.toString());
+    }
+  });
+});
+
+describe("DOMException('abc')", () => {
+  const e = new DOMException("abc");
+
+  it("message property should have the string 'abc'", () => {
+    expect(e.message).toEqual("abc");
+  });
+  it("message property should be the initial value", () => {
+    expect(e.name).toEqual("Error");
+  });
+  it("result of the toString method should be 'Error: abc'", () => {
+    expect(e.toString()).toEqual("Error: abc");
+  });
+  it("throwing and catching exceptions can be done", () => {
+    try {
+      throw new DOMException("abc");
+    } catch (ex) {
+      expect(ex.toString()).toEqual(e.toString());
+    }
+  });
+});
+
+describe("DOMException('abc', 'def')", () => {
+  const e = new DOMException("abc", "def");
+
+  it("message property should have the string 'abc'", () => {
+    expect(e.message).toEqual("abc");
+  });
+  it("name property should have the string 'def'", () => {
+    expect(e.name).toEqual("def");
+  });
+  it("result of the toString method should be 'def: abc'", () => {
+    expect(e.toString()).toEqual("def: abc");
+  });
+  it("throwing and catching exceptions can be done", () => {
+    try {
+      throw new DOMException("abc", "def");
+    } catch (ex) {
+      expect(ex.toString()).toEqual(e.toString());
```

Comment:
Can we also check for stack? If stack is not present we need to change the implementation to inheritance. We'll sort it together :)

Reply:
We are trying to work on error inheritance in the rust implementation. We will modify it accordingly, so we will leave it as it is for now.

Reply:
Sorry for being unclear. This inheritance is in JavaScript user land. For reference see how `buffer.rs` inherits from `Uint8Array` here: https://github.com/awslabs/llrt/blob/7904eb84a34c2cfd20c6a58b46f288d8999a48a1/src/buffer.rs#L195

Reply:
I am sorry. I guess I have been very wrong so far. So this means that I can create a new type on the rust side by inheriting the type that exists on the Javascript side? In other words, I can create an init() like the following and create a DOMException?

```rust
pub fn init<'js>(ctx: &Ctx<'js>) -> Result<()> {
    let ex = ctx.eval::<Object<'js>, &str>(&format!(
        "class {0} extends Error {{}}\n{0}",
        stringify!(DOMException)
    ))?;
    set_prototype(ctx, ex)
}
```
If this is the case, we will need to take some time to review the implementation with reference to the implementation of buffers.rs and its relationship to Uint8Array.

Reply:
Yes, that exactly it! Then you simply define objects on the prototype.

Reply:
Hi @richarddavison ,
Referring to `Buffer.rs`, we have made `Error` in JavaScript user land inherit for `DOMException`.

I think we are getting close to the way it should be, but there are some things that are not working and I left a `TODO:`. Could you please advise?